Element84/filmdrop-ui
# refactor config to load on init

**Related Issue(s):**

- NA

**Proposed Changes:**

1. move config location from `src/assets/config.js` to `./public/config/config.json`
2. change config to be a json object instead of separate exported constants
3. Load config.json into redux on app load once instead of of direct imports
4. Add pre-initialization page to handle and show error (and not render app) if config is missing
5. Change config key names to remove the word `VITE` (leftover from when they were .env vars)
6. Refactor tests for search to be single file now
7. Refactor `default.js` file to only be shared, static values (also clean up dead code here)

BONUS: add leaflet-draw types & fixed a11y error in launchModal

### To Test:

- confirm unit tests pass 🍏 

**Test new config**
- see `/public/config/config.example.json` for new format
- add  director named `config` with file named`config.json` to `/public` folder
> NOTE: see ticket for `config.json` that can be used in testing
- update config params to match existing file constants
- remove old local `config.js` file so it does not get committed by accident if adding new commits
- `npm run start` like normal to load app
- app should have no regressions after changes made (manual spot check to confirm)

**Test no config**
- delete or change name of new `config.json` file
- reload app
- confirm app does not fully load child components
- page should be blank dark mode screen
- confirm error message displays as expected

**PR Checklist:**
